### ISSUE TITLE
Resource and rendering issues with wikis using Lua extensions

### ISSUE THREAD

**Issue by Arcane21** [Orain/ansible-playbook#532]
As noticed on All The Tropes, we get a lot of these and related errors trying to render Lua templates.

Lua error: Cannot create process: fork failed - Cannot allocate memory

This indicates a lack of resources, which needs to evaluated and remedied ASAP.

Also, we do not have the actual Lua backend and Lua Sandbox installed, only the basic framework provided with the extension, which, given our high usage of Lua, is insufficient, and thus the Lua backend and Sandbox need added like they are present on the WMF for added performance.

Instructions for doing so can be found on Scribunto's extension page.

https://www.mediawiki.org/wiki/Extension:Scribunto#Configuration

**Comment by addshore** [comment]
Sounds like either one of the MW instances is running out of mem.

OR

Your lua scripts are using too much mem / are inefficient

I have split adding the new extensions into a different issue

**Comment by labster** [comment]
Parser profiling data | .
------------ | -------------
CPU time usage | 0.000 seconds
Real time usage | 0.393 seconds
Preprocessor visited node count | 69/1000000
Preprocessor generated node count | 0/1000000
Post-expand include size | 3859/2097152 bytes
Template argument size | 0/2097152 bytes
Highest expansion depth | 3/40
Expensive parser function count | 0/100
Lua time usage | 0.050/7.000 seconds
Lua virtual size | 8.2 MB/50 MB
Lua estimated memory usage | 0 B

Yeah so basically fuck you @addshore.  And I say that with love.

Lua is 1/8 of the page render time for a preview, it's not like we're going crazy here.  Not sure how I could reduce memory below 0 B -- maybe quantum computing?  But really, I'm making a tree with 10-200 nodes on it, this is not a case to break out MapReduce for the wiki.  Maybe if Lua had mutable strings it could be more efficient for our case, but you know, feel free to see if you can improve the efficiency of our module code.

**Comment by damuniz** [comment]
woah - we coulda said something a wee bit nicer than "fuck you" - please.

-- 
Thank You,
Dusti Muniz

**Comment by labster** [comment]
Yeah, well I've been dealing with similar attitudes at work where I really can't say such things.  But I think you're overreacting @damuniz, given the tone of the note.

Though this has been a recurring problem for a long time, though lua only seems to die during even-numbered months.  It's solved, then it comes back with basically no code changes on the user side.  If one lua call per page is too much for the infrastructure to handle, I could rewrite it as a parser extension in PHP -- though on en.wikipedia the opinion seems to be the Lua is faster than templates, which is why I wrote our current templates in the first place.

**Comment by damuniz** [comment]
I'm not overreacting to anything really - I'm not even reacting. I'm just
asking that we say things a little nicer, considering that this is a public
forum and we're *all *volunteers here - including Addshore. I know you
didn't mean it harshly and that you were saying so in jest, I just don't
want animosity around here, please. :)

That's all dears.

-- 
Thank You,
Dusti Muniz

**Comment by labster** [comment]
Ease off the passive aggression, Dusti, and I think we'll be fine.  This is a forum for discussion of technical issues -- what we're saying should be backed up by facts rather than opinions.  That's all I ask.

**Comment by damuniz** [comment]
???? I swear there's no aggression Brent. Zero. Zip. None. Passive or
active. I promise. I was only worried about the "fuck you" part, and I
addressed it and you responded in kind.

-- 
Thank You,
Dusti Muniz

**Comment by Southparkfan** [comment]
This leak of memory [already caused downtime](https://tech.orain.org/wiki/Incidents/2015-02-15) after this issue was filed.

**Comment by addshore** [comment]
So, after reading all of the deleted comments (and bits of comments) in my emails I'll follow with this...

> Yeah so basically fuck you @addshore. And I say that with love.

Wow, how kind, I feel very motivated to try and help you solve this issue...

> feel free to see if you can improve the efficiency of our module code.

Which module?

Anyway, I'll link this back to the previous issues (that I have not seen before https://github.com/Orain/ansible-playbook/issues/412 )
